container-core/src/main/java/com/yahoo/container/handler/ClustersStatus.java

```diff
@@ -37,11 +39,15 @@ public ClustersStatus() { }
     /** The status of clusters, when known. Note that clusters may exist for which there is no knowledge yet. */
     private final Map<String, Boolean> clusterStatus = new HashMap<>();
 
-    public void setContainerHasClusters(boolean containerHasClusters) {
+    /** Sets the current clusters of this container */
+    public void setClusters(Set<String> clusters) {
         synchronized (mutex) {
-            this.containerHasClusters = containerHasClusters;
-            if ( ! containerHasClusters)
-                clusterStatus.clear(); // forget container clusters which was configured away
+            this.containerHasClusters = clusters.size() > 0;
```

Comment:
Use isEmpty ?

Reply:
I actually think this is slightly clearer by avoiding negation.